### PR TITLE
SONARPY-1299 Ensure rules unit tests can run on IPython files

### PR DESCRIPTION
**Pull request by maksim-grebeniuk-sonarsource**

### CODE REVIEW

File: python-checks/src/test/java/org/sonar/python/checks/quickfix/PythonQuickFixVerifier.java [hunk 1]
```diff
@@ -49,8 +49,32 @@ private PythonQuickFixVerifier() {
...
+
+  public static void verifyNoQuickFixes(PythonCheck check, String codeWithIssue) {
+    verifyNoQuickFixes(PythonQuickFixVerifier::createPythonVisitorContext, check, codeWithIssue);
+  }
+
+  public static void verifyQuickFixMessages(PythonCheck check, String codeWithIssue, String... expectedMessages) {
+    verifyQuickFixMessages(PythonQuickFixVerifier::createPythonVisitorContext, check, codeWithIssue, expectedMessages);
+  }
+
+  public static void verifyIPython(PythonCheck check, String codeWithIssue, String... codesFixed) {
```
Comment (guillaume-dequenne-sonarsource) [inline python-checks/src/test/java/org/sonar/python/checks/quickfix/PythonQuickFixVerifier.java:63]: Should this be `verifyIPythonQuickFixes` for consistency?

**Review by guillaume-dequenne-sonarsource — CHANGES_REQUESTED** [review]
Looking good, left small comments.
I'm not sure I see the value in the quick fix tests that were implemented.
That said, I think the approach to make IPython quick fixes testable is nice.

File: python-checks/src/test/resources/checks/deadStore.ipynb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+def foo():
```
[the remark this answers is not in the document]
Comment (maksim-grebeniuk-sonarsource) [reply python-checks/src/test/resources/checks/deadStore.ipynb:1]: done

File: python-frontend/src/main/java/org/sonar/python/TestPythonVisitorRunner.java [hunk 3]
```diff
@@ -62,25 +62,30 @@ public static PythonVisitorContext createContext(File file, @Nullable File worki
...
   }
 
   public static ProjectLevelSymbolTable globalSymbols(List<File> files, File baseDir) {
     ProjectLevelSymbolTable projectLevelSymbolTable = new ProjectLevelSymbolTable();
     for (File file : files) {
-      TestPythonFile pythonFile = new TestPythonFile(file);
-      AstNode astNode = PythonParser.create().parse(pythonFile.content());
-      FileInput astRoot = new PythonTreeMaker().fileInput(astNode);
+      var pythonFile = new TestPythonFile(file);
+      var astRoot = parseFile(pythonFile);
```
[the remark this answers is not in the document]
Comment (maksim-grebeniuk-sonarsource) [reply python-frontend/src/main/java/org/sonar/python/TestPythonVisitorRunner.java:74]: done

File: python-checks/src/test/java/org/sonar/python/checks/DeadStoreCheckTest.java [hunk 4]
```diff
@@ -55,6 +55,23 @@ public void assignment_on_single_line() {
...
 
+  @Test
+  public void assignment_on_single_line_ipython() {
+    String codeWithIssue = code(
+      "def foo():",
+      "    x = 42",
+      "    x??",
+      "    x = 0",
+      "    print(x)");
+    String codeFixed = code(
```
[the remark this answers is not in the document]
Comment (maksim-grebeniuk-sonarsource) [reply python-checks/src/test/java/org/sonar/python/checks/DeadStoreCheckTest.java:66]: test removed

File: python-checks/src/test/java/org/sonar/python/checks/DeadStoreCheckTest.java [hunk 5]
```diff
@@ -161,6 +178,18 @@ public void function_call_as_assigned_value() {
     PythonQuickFixVerifier.verifyNoQuickFixes(check, codeWithIssue);
   }
 
+  @Test
+  public void function_call_as_assigned_value_ipython() {
```
[the remark this answers is not in the document]
Comment (maksim-grebeniuk-sonarsource) [reply python-checks/src/test/java/org/sonar/python/checks/DeadStoreCheckTest.java:182]: test removed

**Review by guillaume-dequenne-sonarsource — APPROVED** [review]
LGTM.

**Pull request merged**